src/sentry/static/sentry/app/views/performance/compare/utils.tsx

```diff
@@ -318,17 +323,62 @@ function createChildPairs({
   };
 }
 
+function jaroSimilarity(thisString: string, otherString: string): number {
+  // based on https://winkjs.org/wink-distance/string-jaro-winkler.js.html
+  // and https://en.wikipedia.org/wiki/Jaro%E2%80%93Winkler_distance
+  if (thisString === otherString) {
+    return 1;
+  }
+
+  let jaroDistance: number = jaro(thisString, otherString).distance;
+
+  const scalingFactor = 0.1;
+  const boostThreshold = 0.3;
+
+  if (jaroDistance > boostThreshold) {
+    return 1 - jaroDistance;
+  }
+
+  const pLimit = Math.min(thisString.length, otherString.length, 4);
+  let l = 0;
+
+  for (let i = 0; i < pLimit; i += 1) {
+    if (thisString[i] === otherString[i]) {
+      l += 1;
+    } else {
+      break;
+    }
+  }
+
+  jaroDistance -= l * scalingFactor * jaroDistance;
+  return 1 - jaroDistance;
+}
+
 function matchableSpans({
   baselineSpan,
   regressionSpan,
 }: {
   baselineSpan: SpanType;
   regressionSpan: SpanType;
-}): boolean {
+}): number {
   const opNamesEqual = baselineSpan.op === regressionSpan.op;
-  const descriptionsEqual = baselineSpan.description === regressionSpan.description;
 
-  return opNamesEqual && descriptionsEqual;
+  if (!opNamesEqual) {
+    return 0;
+  }
+
+  // remove whitespace and convert string to lower case as the individual characters
+  // adds noise to the edit distance function
+  const baselineDescription = (baselineSpan.description || '')
+    .replace(/\s+/g, '')
+    .toLowerCase();
+  const regressionDescription = (regressionSpan.description || '')
+    .replace(/\s+/g, '')
+    .toLowerCase();
+
+  const score = jaroSimilarity(baselineDescription, regressionDescription);
+
+  return score >= 0.8 ? score : 0;
```

Comment:
Let’s pull this up into a constant as well? I can see that is can be something we change.

Reply:
Fixed in https://github.com/getsentry/sentry/pull/20224/commits/8a9ed42d953f4463c778a75edb4ef215f79a00ff